### DISCUSSION TITLE
New context API doesn't make sense?

### DISCUSSION THREAD

**Discussion by Anthonyzou** [ExpediaGroup/graphql-kotlin#1268] [Q&A] [answered]
So I have instrumentation that accesses the context but the new API always returns an empty map.

<img width="937" alt="Screen Shot 2021-10-07 at 2 20 13 PM" src="https://user-images.githubusercontent.com/1567160/136456342-9c933a2d-0829-44a2-bc0d-19532c1117fe.png">

I am using the ktor web server and I have a factory to generate the context but how can I possibly set the context using the new API?

<img width="931" alt="Screen Shot 2021-10-07 at 2 20 40 PM" src="https://user-images.githubusercontent.com/1567160/136456393-c57549f4-420a-4927-a433-ae508ba8e947.png">
>

**Comment by smyrick** [comment]
Your question is specific to the new graphql-java 17 context object. You can find out more info about that here: https://github.com/graphql-java/graphql-java/pull/2368

**Reply by Anthonyzou** [reply]
That kind of helped, but I figured out the issue. It is not possible to remove the `.context` call from `ExecutionInput`.

<img width="731" alt="Screen Shot 2021-10-07 at 4 35 39 PM" src="https://user-images.githubusercontent.com/1567160/136471450-ca621b7d-2c7c-492c-859c-ce60ff66019c.png">

Basically without the `.context` call the library will think `GQLContext` is an input type instead of a context to inject. 
How will the context get injected in the future if it becomes deprecated?


<img width="382" alt="Screen Shot 2021-10-07 at 5 00 13 PM" src="https://user-images.githubusercontent.com/1567160/136473571-4a58d8f4-b28e-4bee-938f-5de2e8b61238.png">

**Reply by smyrick** [reply]
We will most likely deprecate our `GraphQLContext` interface in favor of just getting the `DataFetchingEnvironment` but we also wanted to see how the [spring-graphql](https://spring.io/projects/spring-graphql) integration would look as we would like to adopt that over our current custom implementation. Once we know how we can best have the generator hook into that project, we will be making more changes.

In the meantime, if you wanted to use the new context map instead of the deprecated context object, you can do this


```kt
fun test(dfe: DataFetchingEnvironment): Boolean {
  return dfe.graphqlContext.get("name") == null
}
```

**Comment by dariuszkuc** [comment, marked as answer]
Support for new GraphQL context is added through https://github.com/ExpediaGroup/graphql-kotlin/pull/1292
